Find a functionally equivalent code:android.text.SpannableStringInternal.setSpan:COMMENT
Method Modifier: 
Comment:/* package */

@@ -29,37 +29,36 @@
         }
     }
     if (mSpanCount + 1 >= mSpans.length) {
-        int newsize = ArrayUtils.idealIntArraySize(mSpanCount + 1);
-        Object[] newtags = new Object[newsize];
-        int[] newdata = new int[newsize * 3];
+        Object[] newtags = ArrayUtils.newUnpaddedObjectArray(GrowingArrayUtils.growSize(mSpanCount));
+        int[] newdata = new int[newtags.length * 3];
         System.arraycopy(mSpans, 0, newtags, 0, mSpanCount);
         System.arraycopy(mSpanData, 0, newdata, 0, mSpanCount * 3);
         mSpans = newtags;

